pkg/didcomm/dispatcher/outbound.go

```diff
@@ -266,11 +319,25 @@ func (o *OutboundDispatcher) addTransportRouteOptions(req []byte, des *service.D
 	return req, nil
 }
 
-// TODO - inject MediaTypeProfile selection strategy.
 func mediaTypeProfile(des *service.Destination) string {
 	mt := ""
+	v1MT := ""
+
 	if len(des.MediaTypeProfiles) > 0 {
-		mt = des.MediaTypeProfiles[0]
+		for _, mtp := range des.MediaTypeProfiles {
+			switch mtp {
+			case transport.MediaTypeV1PlaintextPayload, transport.MediaTypeRFC0019EncryptedEnvelope:
+				// overridable with higher priority media type.
+				if v1MT == "" {
+					v1MT = mtp
+				}
+			case transport.MediaTypeV1EncryptedEnvelope, transport.MediaTypeV2EncryptedEnvelopeV1PlaintextPayload:
+				v1MT = mtp
+			case transport.MediaTypeV2EncryptedEnvelope, transport.MediaTypeV2PlaintextPayload:
+				// V2 is highest priority, if found use it directly.
+				return mtp
+			}
+		}
 	}
 
 	return mt
```

Comment:
`mt` is never set to anything, do you mean to return `v1MT` instead?

Reply:
indeed, `v1MT` is unnecessary, I've updated to the code to use `mt` instead.